template/registry_datetime.go

```diff
@@ -236,5 +270,14 @@ func datetimeFunc(input any, options Options, locale language.Tag) (any, error)
 
 	value = value.In(opts.TimeZone)
 
+	if opts.Year != "" {
```

Comment:
It will fail earlier in parseDatetimeOptions

Reply:
Should we correct the flow? We always compute `TimeStyle` and `Year`, and I understand they both can't be set at the same time, right? `switch`?

Reply:
Well, they both won't be computed, either one or the other because if both are set, the conflicting checking function will fail. I think better than a switch would be checking for some done calculations (e.g. if `layout` is not empty string) just before field options calculations (this line) for early exit.